WalletWasabi/Transactions/TransactionBuilding/SmartCoinSelector.cs

```diff
@@ -29,23 +30,54 @@ public IEnumerable<ICoin> Select(IEnumerable<ICoin> coins, IMoney target)
 				throw new InsufficientBalanceException(targetMoney, available);
 			}
 
-			// Group coins by scriptPubKey in order to treat them all as one.
-			var coinsByScriptPubKey = UnspentCoins
+			var uniqueClusters = UnspentCoins
+				.Select(coin => coin.Clusters)
+				.Distinct();
+
+			var clusters = (uniqueClusters.Count() < 10)
+				? uniqueClusters
+					.CombinationsWithoutRepetition(ofLength: 1, upToLength: 6)
+					.Select(clusterCombination => UnspentCoins.Where(coin => clusterCombination.Contains(coin.Clusters)))
```

Comment:
Contains shouldn't work either, because `Cluster` doesn't imlement iequality and friends (I guess gethashscode is needed for that)

Reply:
```c#
var a = new { s="hello" };
var b = new { s="bye" };
var c = a;
new[]{ a, b, c }.Contains(b)
true
```